src/loaders/jobs/taskfarmer/taskfarmer_task_mgr.py

```diff
@@ -0,0 +1,357 @@
+        N          0           N             N/A              FAILED (If a job is cancelled, it will be treated as failed.)
+        N          0           Y             CG               CANCELLED
+        """
+
+        # Once all tasks are completed, NERSC generates this file.
+        # (ref: https://docs.nersc.gov/jobs/workflow/taskfarmer/#output)
+        done_file = os.path.join(self.job_dir, NERSC_SLURM_DONE_FILE)
+        if os.path.isfile(done_file):
+            return JobStatus.COMPLETED
+
+        # check job status using squeue command
+        std_out_file, std_err_file, exit_code = tf_common.run_nersc_command(
+            ['squeue', '-j', str(job_id)], self.job_dir, log_file_prefix='squeue', check_return_code=False)
+
+        if exit_code != 0:
+            # job finished without creating the done file indicating the job is failed
+            return JobStatus.FAILED
+
+        with open(std_out_file, "r") as std_out, open(std_err_file, "r") as std_err:
+            squeue_out = std_out.read().strip()
+            if str(job_id) in squeue_out:
+                job_str = squeue_out.splitlines()[1]
+                parsed_status = job_str.split()[4]
+                if parsed_status == 'PD':
+                    return JobStatus.PENDING
+                elif parsed_status == 'R':
+                    return JobStatus.RUNNING
+                elif parsed_status == 'CG':
+                    return JobStatus.CANCELLED
+                else:
+                    raise ValueError(f"Unrecognized job status: {squeue_out}")
+            else:
+                return JobStatus.FAILED
+
+    def _append_task_info(self, task_info):
+        """
+        Append the task information to the task info file
+        """
+
+        if not all(key in task_info for key in REQUIRED_TASK_INFO_KEYS):
+            raise ValueError(f"task_info must contain all keys: {REQUIRED_TASK_INFO_KEYS}")
+
+        task_info.update({'kbase_collection': self.kbase_collection, 'load_ver': self.load_ver, 'tool': self.tool,
+                          'source_data_dir': self.source_data_dir})
+
+        # Append the new record to the file
+        task_info_file = self._get_task_info_file()
+        with open(task_info_file, 'a') as writer:
+            fcntl.flock(writer.fileno(), fcntl.LOCK_EX)
+            json.dump(task_info, writer)
+            writer.write('\n')
+            fcntl.flock(writer.fileno(), fcntl.LOCK_UN)
+
+    def _task_exists(self):
+        """
+        Check if the task for kbase_collection, load_ver, tool has been submitted.
+        """
+
+        if self.force_run:
+            # The job directory is recreated as part of the initialization process.
+            job_dir_exists = True
+        else:
+            # check if the job directory exists and is not empty
+            job_dir_exists = os.path.isdir(self.job_dir) and os.listdir(self.job_dir)
```

Comment:
I'm still having trouble with the _task_exists logic. I'm not sure I understand why the directory is taken into account. If the job exists, shouldn't that be the gold standard? 

For example, if someone manually deletes the job dir and then runs the task submission script again, the new jobs will be scheduled while the old jobs are still running, IIUC

Reply:
👍 yea. you are probably right. removed job_dir check.

Reply:
:+1: